extensions/spring-xd-extension-batch/src/main/java/org/springframework/batch/step/tasklet/x/AbstractProcessBuilderTasklet.java

```diff
@@ -102,34 +163,37 @@ public RepeatStatus execute(StepContribution contribution, ChunkContext chunkCon
 			logger.error(msg);
 		}
 		finally {
-			printLog(commandName, commandLog, exitCode);
-			StringBuilder firstException = new StringBuilder();
-			if (exitCode != 0) {
-				for (String line : commandLog) {
-					if (firstException.length() == 0) {
-						if (line.contains("Exception")) {
-							firstException.append(line).append("\n");
-						}
-					}
-					else {
-						if (line.startsWith("\t")) {
-							firstException.append(line).append("\n");
-						}
-						else {
-							break;
-						}
-					}
+			if (p != null) {
+				p.destroy();
+			}
+			commandOut = getProcessOutput(out);
+			commandErr = getProcessOutput(err);
+			printLog(commandName, commandOut, commandErr);
```

Comment:
We load 10000 first lines and the main purpose is to store that in the step execution. We also use a selection of these for debug logging and error logging in the XD container log. if there is a better way, I'm fine with that.